crypto/http/http_client.c

```diff
@@ -1056,11 +1060,10 @@ ASN1_VALUE *OSSL_HTTP_get_asn1(const char *url,
         ERR_raise(ERR_LIB_HTTP, ERR_R_PASSED_NULL_PARAMETER);
         return NULL;
     }
-    if ((mem = OSSL_HTTP_get(url, proxy, no_proxy, bio, rbio, bio_update_fn,
-                             arg, headers, maxline, max_resp_len, timeout,
-                             expected_ct, 1 /* expect_asn1 */))
-        != NULL)
-        resp = BIO_mem_d2i(mem, rsp_it);
+    mem = OSSL_HTTP_get(url, proxy, no_proxy, bio, rbio, bio_update_fn,
+                        arg, headers, maxline, max_resp_len, timeout,
+                        expected_ct, 1 /* expect_asn1 */);
+    resp = BIO_mem_d2i(mem /* may be NULL */, rsp_it);
```

Comment:
In case you might be knowingly passing NULL into  BIO_mem_d2i() that function should not raise ERR_R_PASSED_NULL_PARAMETER error.

Reply:
Good point - so removed the `ERR_raise(ERR_LIB_HTTP, ERR_R_PASSED_NULL_PARAMETER)` call in the function.